apps/api/src/bank-transactions-file/helpers/parser.ts

```diff
@@ -39,3 +46,11 @@ export function parseBankTransactionsFile(
   })
   return accountMovements
 }
+
+function isBillingNamePresent(movementFunctionalType: string, billingName: string) {
+  return !(
+    movementFunctionalType === cashDepositBGString &&
+    billingName['$'] !== undefined &&
```

Comment:
It's a bit strange to expect `billingName` to be string in the fn definition and then to use it array here.

Please improve the types of the input.

Reply:
This is a tricky one, because:
When we have   <OppositeSideName>Firstname Lastnameee</OppositeSideName> we pass string to the function.
On the other hand  <OppositeSideName nil="true" /> is passed as Json object.
I used the following option:
```
    string | { [x: string]: unknown }
```

I'm open to any suggestions, if there is a better way to improve that.

Reply:
@Nnachevvv  The update looks great, brings a lot more context, and clears up the duality of the types.

Thanks for the contribution!

Reply:
@Nnachevvv  After some more thinking about this piece of code i figured out the intent was not super clear as we have two separate functionality in the same function right now. We can separate them as:


```tsx
const oppositeSide = items[item].AccountMovement[movement].OppositeSideName[0]
if (isCashDeposit(movementFunctionalType) && !isBillingNameNil(oppositeSide)) {
  payment.billingName = oppositeSide
}

function isCashDeposit(movementFunctionalType: string): boolean {
  return movementFunctionalType === cashDepositBGString
}
function isBillingNameNil(
  billingName: string | { [x: string]: unknown },
): boolean {
  return (
    billingName['$'] !== undefined &&
    billingName['$']['nil']
  )
}
```

Reply:
I agree with you. It looks better , I will make a new PR with the change. Thanks!